src/main/java/ch/njol/skript/expressions/ExprReplace.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.SkriptConfig;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.ReplacementTypes;
+import ch.njol.util.Kleenean;
+
+
+@Name("Replace")
+@Description("Replaces all occurrences of a given text with another text and returns the replaced text.")
+@Examples({"on chat:",
+		"\tset {_hey} to replace all \"hello\" in message with \"hey\" #this will replace all the values without changing the message",
+		"\tsend {_hey}",
+		"set {_no} to case-sensitive replace first \"yes\" in \"Yes, yes\" with \"no\" #Only the second yes gets replaced with no"
+})
+@Since("INSERT VERSION")
+public class ExprReplace extends SimpleExpression<String> {
+	static {
+		Skript.registerExpression(ExprReplace.class, String.class, ExpressionType.COMBINED,
+				"[(4¦case-sensitive)] replace (1¦first|2¦last|0¦all|every|) %strings% with %string% in %string%",
```

Comment:
I think the old EffReplace case sensitivity pattern would work better here. Up for discussion though.

Reply:
Yeah, I thought so too